common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -668,14 +684,110 @@ private static boolean isAndroid0() {
             logger.debug("Platform: Windows");
         }
         return windows;
     }
 
+    private static boolean isLinux0() {
+        boolean linux = isOs("linux");
+        if (linux) {
+            logger.debug("Platform: Linux");
+        }
+        return linux;
+    }
+
+    private static boolean isOsx0() {
+        boolean osx = isOs("os x") || isOs("osx");
+        if (osx) {
+            logger.debug("Platform: OSX");
+        }
+        return osx;
+    }
+
+    private static int cacheLineSize0() {
+        return AccessController.doPrivileged(new PrivilegedAction<Integer>() {
+            @Override
+            public Integer run() {
+                if (IS_LINUX) {
+                    BufferedReader in = null;
+                    try {
+                        in = new BufferedReader(new InputStreamReader(new FileInputStream("/proc/cpuinfo")));
+                        String line;
+                        while ((line = in.readLine()) != null) {
+                            if (line.startsWith("cache_alignment")) {
+                                String[] parts = StringUtil.split(line, ':');
+                                return Integer.parseInt(parts[1].trim());
+                            }
+                        }
+                    } catch (Throwable cause) {
+                        logger.debug("Failed to retrieve cache line size", cause);
+                        return -1;
+                    } finally {
+                        if (in != null) {
+                            try {
+                                in.close();
+                            } catch (IOException ignore) {
+                                // ignore on close
+                            }
+                        }
+                    }
+                } else if (IS_OSX) {
+                    Process p = null;
+                    BufferedReader in = null;
+                    try {
+                        p = Runtime.getRuntime().exec(new String[] { "sysctl", "hw.cachelinesize" });
+                        in = new BufferedReader(new InputStreamReader(p.getInputStream(), CharsetUtil.US_ASCII));
+                        String line = in.readLine();
+                        in.close();
+                        try {
+                            int exitCode = p.waitFor();
+                            if (exitCode != 0) {
+                                logger.debug("Failed to retrieve cache line size," +
+                                        " 'sysctl hw.cachelinesize' did exit with code " + exitCode);
+                                return -1;
+                            }
+                        } catch (InterruptedException e) {
+                            Thread.currentThread().interrupt();
+                        }
+                        if (line != null) {
+                            String[] parts = StringUtil.split(line, ':');
+                            return Integer.parseInt(parts[1].trim());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method io.netty.util.internal.PlatformDependent$1.run() converts String to boxed primitive using excessive boxing [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOXING_VALUEOF)